common/src/main/java/org/jboss/eap/qe/ts/common/docker/Docker.java

```diff
@@ -0,0 +1,193 @@
+package org.jboss.eap.qe.ts.common.docker;
+
+import org.fusesource.jansi.Ansi;
+import org.junit.rules.ExternalResource;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ *
+ * Utility class for starting docker containers. This class allow to start any docker container before the test.
+ *
+ * Intended to be used as a JUnit {@link org.junit.ClassRule ClassRule}.
+ *
+ * Example of usage:
+ *
+ * import org.jboss.arquillian.container.test.api.RunAsClient;
+ * import org.jboss.arquillian.junit.Arquillian;
+ * import org.jboss.arquillian.junit.InSequence;
+ * import org.jboss.eap.qe.ts.common.docker.Docker;
+ * import org.jboss.eap.qe.ts.common.docker.DockerContainers;
+ * import org.junit.ClassRule;
+ * import org.junit.Test;
+ * import org.junit.runner.RunWith;
+ *
+ * @RunAsClient
+ * @RunWith(Arquillian.class)
+ * public class TestDocker {
+ *
+ * // Docker container will be started in @BeforeClass phase - see list and current configuration of current {@link DockerContainers}
+ * @ClassRule
+ * public static Docker jaegerContainer=DockerContainers.jaeger();
+ *
+ *
+ * @Test
+ * public void test()  {
+ *  // test something with docker container...
+ * }
+ * }
+ */
+public class Docker extends ExternalResource {
+    private final String uuid;
+    private final String name;
+    private final String image;
+    private final List<String> ports = new ArrayList<>();
+    private final Map<String, String> environmentVariables = new HashMap<>();
+    private final List<String> commandArguments = new ArrayList<>();
+    private String awaitedLogLine;
+    private long waitTimeoutInMillis = 600_000; // 10 minutes
+
+    private final ExecutorService outputPrinter = Executors.newSingleThreadExecutor();
+
+    public Docker(String name, String image) {
+        this.uuid = UUID.randomUUID().toString();
+        this.name = name;
+        this.image = image;
+    }
+
+    public Docker waitForLogLine(String logWait) {
+        this.awaitedLogLine = logWait;
+        return this;
+    }
+
+    public Docker setWaitForLogLineTimeout(long timeoutInMillis) {
+        this.waitTimeoutInMillis = timeoutInMillis;
+        return this;
+    }
+
+    public Docker addPort(String port) {
+        this.ports.add(port);
+        return this;
+    }
+
+    public Docker addEnvVar(String key, String value) {
+        environmentVariables.put(key, value);
+        return this;
+    }
+
+    public Docker addCmdArg(String commandArgument) {
```

Comment:
I see lots of setters. Are all of them mandatory? It should be documented which are mandatory. Or create some builder with proper javadoc (where we can see defaults and so on)